plugins/ComposeAuth/src/commonMain/kotlin/io/github/jan/supabase/compose/auth/ComposeAuth.kt

```diff
@@ -0,0 +1,63 @@
+package io.github.jan.supabase.compose.auth
+
+import io.github.jan.supabase.SupabaseClient
+import io.github.jan.supabase.gotrue.gotrue
+import io.github.jan.supabase.gotrue.providers.Google
+import io.github.jan.supabase.gotrue.providers.builtin.IDToken
+import io.github.jan.supabase.plugins.SupabasePlugin
+import io.github.jan.supabase.plugins.SupabasePluginProvider
+
+
+sealed interface ComposeAuth : SupabasePlugin {
+
+    data class Config(
+        var loginConfig: LoginConfig? = null
+    ) : SupabasePlugin
+
+    companion object : SupabasePluginProvider<Config, ComposeAuth> {
+
+        override val key: String = "composeauth"
+
+        const val API_VERSION = 1
+
+        override fun create(supabaseClient: SupabaseClient, config: Config): ComposeAuth {
+            return ComposeAuthImpl(config, supabaseClient)
+        }
+
+        override fun createConfig(init: Config.() -> Unit): Config {
+            return Config().apply(init)
+        }
+    }
+
+    suspend fun fallbackLogin()
+}
+
+class ComposeAuthImpl(
+    val config: ComposeAuth.Config,
+    val supabaseClient: SupabaseClient,
+) : ComposeAuth {
+
+    val apiVersion = ComposeAuth.API_VERSION
+
+    val pluginKey = ComposeAuth.key
+
+    suspend fun loginWithGoogle(idToken: String) {
+        val nonce = if(config.loginConfig == null || config.loginConfig !is GoogleLoginConfig) null else (config.loginConfig as GoogleLoginConfig).nonce
```

Comment:
```suggestion
        val nonce = (config.loginConfig as? GoogleLoginConfig)?.nonce
```
I think you can make this a little bit easier